source/merge.ts

```diff
@@ -0,0 +1,93 @@
+import {URL} from 'url';
+import is from '@sindresorhus/is';
+// XO doesn't recognize `export type` for some reason.
+// eslint-disable-next-line import/named
+import knownHookEvents, {Hooks, HookType, HookEvent} from './known-hook-events';
+
+// TODO: Use the Got options-object types.
+interface Options {
```

Comment:
The extranous `Options` interface is meant to be merged with the interface that was introduced in https://github.com/sindresorhus/got/pull/708 into a single shared interface.